src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -230,12 +230,13 @@ private static void mergeHeaderlessBamShards(String outputFile, SAMFileHeader he
         // terminating end-of-file marker. Note that this has the side effect of being ever-so-slightly less efficient
         // than writing a BAM in one go because the last block of each file isn't completely full.
 
-        final Configuration conf = new Configuration();
-        final FileSystem fs = FileSystem.get(conf);
         final String outputParentDir = outputFile.substring(0, outputFile.lastIndexOf('/') + 1);
         // First, check for the _SUCCESS file.
         final String successFile = outputFile + "/_SUCCESS";
-        if (!fs.exists(new Path(successFile))) {
+        final Path successPath = new Path(successFile);
+        final Configuration conf = new Configuration();
+        final FileSystem fs = successPath.getFileSystem(conf);
```

Comment:
So it's also possible we want to pass in the actual configuration instead of using a newly generated one.  I don't know if that is important though.